### PR TITLE
test(flattenObject): change using constant date in test case

### PR DESCRIPTION
**Pull request by choi2021**
I have made a PR regarding the [take(compat) function](https://github.com/toss/es-toolkit/pull/252), but it failed due to issues in the flattenObject test case. 

The failure was caused by different date objects being used for comparison in the test cases. 

To fix this, I have modified the test cases to use a consistent date object for comparisons, ensuring that the date objects being compared are identical.

<table>
 <tr>
  <th>AS-IS </th>
</tr>
<tr>
  <td>

</td>
</td>
</tr>
 <tr>
  <th>TO-BE</th>
</tr>
<tr>
  <td> 
스크린샷 2024-07-15 오전 12 40 52
</td>
</tr>
</table>

### CODE REVIEW

File: src/object/flattenObject.spec.ts [hunk 1]
```diff
@@ -13,6 +13,8 @@ describe('flattenObject', function () {
       'a.b': 'yay'
     })
 
+    const date= new Date();
```
Comment (fvsch) [inline src/object/flattenObject.spec.ts:16]: Formatting: missing a space before the `=` sign.

File: src/object/flattenObject.spec.ts [hunk 2]
```diff
@@ -32,11 +34,13 @@ describe('flattenObject', function () {
       'a.b.boolean': true,
       'a.b.null': null,
       'a.b.undefined': undefined,
-      'a.b.date': new Date(),
+      'a.b.date': date,
     })
   })
 
   it('flattens multiple keys', () => {
+    const date= new Date();
```
Comment (fvsch) [inline src/object/flattenObject.spec.ts:42]: Formatting: same issue here.

**Review by fvsch — APPROVED** [review]
This looks good to me.

There are a couple code formatting issues. This could be fixed by running Prettier on the whole file (`yarn prettier --write src/object/flattenObject.spec.ts`), but that would create a larger diff because the file has mixed formatting currently.

PS: I approved this PR even though I’m not a maintainer. I’m not sure what this project's policy is for review by non-maintainers. Is it fine to approve, or better to just comment?

File: src/object/flattenObject.spec.ts (hunk 1 above)
Comment (raon0211) [reply src/object/flattenObject.spec.ts:16]:
```suggestion
    const date = new Date();
```

File: src/object/flattenObject.spec.ts (hunk 2 above)
Comment (raon0211) [reply src/object/flattenObject.spec.ts:42]:
```suggestion
    const date = new Date();
```

**Review by raon0211 — APPROVED** [review]
Thanks!

**Pull request merged**